src/js/binaryen.js-post.js

```diff
@@ -3043,8 +3043,11 @@ Module['setAllowInliningFunctionsWithLoops'] = function(value) {
 
 // Expression wrappers
 
-// Makes a wrapper class with the specified static members while
-// automatically deriving instance methods and accessors.
+// Private symbol used to store the underlying C-API pointer of a wrapped object.
+const thisPtr = Symbol();
```

Comment:
This will require a somewhat new JS VM, I think? Wouldn't that limit our compatibility?

Reply:
Hmm, thought that'd be available due to switching the post JS to ES6 recently, or does Closure Compiler transform to ES5?

Reply:
As I remember closure currently accept ES6 input and output same ES6.

Reply:
According to MDN, `Symbol` is supported since

* Chrome 38
* Edge 12
* Firefox 36
* Safari 9
* Opera 25
* Node.js 0.12

Destructuring assignments, which we already use, are supported since

* Chrome 49
* Edge 14
* Firefox 41
* Safari 8
* Opera 36
* Node.js 6

All well beyond symbols, except Safari. However, Safari supports arrow functions since 10, so should have broken already if there was a problem. Based on this, usage of `Symbol` appears safe to me.

Reply:
Sounds good, thanks for checking!